3rdparty/sparse-matrix/sparse_matrix.cc

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <mkl_spblas.h>
+#include "sparse_matrix.h"
+
+
+
+bool mkl_DotCsrDnsDns(SP_INT64* rows_start, SP_INT64* col_indx,
```

Comment:
Why does it land in the 3rdparty directory, instead of src/operator/?

Reply:
Is there future plan to support other sparse matrix ops, like dense + csr, csr + csr, csr * csr (elemwise)?

Reply:
@eric-haibin-lin 
If you requests, e.g. mkl has 
mkl_sparse_spmm s, d, c, z Computes the product of two sparse matrices and stores
the result as a sparse matrix
why this patch is merge first is because Wide & Deep use sparse dot dense matrix

Reply:
> Why does it land in the 3rdparty directory, instead of src/operator/?

This is the quick solution for the wide deep. To move the code into src/operator, need to change code in mshadow and more tests are needed. I will do it as the next step.

Reply:
Thanks for the explanation. Why mshadow has to be changed?

Reply:
mshadw use 32bit interface , below is the link for mkl settings
https://software.intel.com/en-us/articles/intel-mkl-link-line-advisor
and the make should change 
diff --git a/make/mshadow.mk b/make/mshadow.mk
index 86155ea..d0b731e 100644
--- a/make/mshadow.mk
+++ b/make/mshadow.mk
@@ -82,7 +82,7 @@ ifneq ($(USE_INTEL_PATH), NONE)
 		MSHADOW_LDFLAGS += -L$(USE_INTEL_PATH)/compiler/lib/intel64
 		MSHADOW_LDFLAGS += -L$(USE_INTEL_PATH)/lib/intel64
 	endif
-	MSHADOW_CFLAGS += -I$(USE_INTEL_PATH)/mkl/include
+	MSHADOW_CFLAGS += -I$(USE_INTEL_PATH)/mkl/include -DMKL_ILP64 -m64
 endif
 ifneq ($(USE_STATIC_MKL), NONE)
 ifeq ($(USE_INTEL_PATH), NONE)
@@ -90,7 +90,7 @@ ifeq ($(USE_INTEL_PATH), NONE)
 else
 	MKLROOT = $(USE_INTEL_PATH)/mkl
 endif
-	MSHADOW_LDFLAGS += -L${MKLROOT}/../compiler/lib/intel64 -Wl,--start-group ${MKLROOT}/lib/intel64/libmkl_intel_lp64.a ${MKLROOT}/lib/intel64/libmkl_core.a ${MKLROOT}/lib/intel64/libmkl_intel_thread.a -Wl,--end-group -liomp5 -ldl -lpthread -lm
+	MSHADOW_LDFLAGS += -L${MKLROOT}/../compiler/lib/intel64 -Wl,--start-group ${MKLROOT}/lib/intel64/libmkl_intel_ilp64.a ${MKLROOT}/lib/intel64/libmkl_core.a ${MKLROOT}/lib/intel64/libmkl_intel_thread.a -Wl,--end-group -liomp5 -ldl -lpthread -lm
 else

Reply:
Look forward to the followup PR